src/vm.rs

```diff
@@ -547,23 +580,10 @@ impl<T: GuestStagePageTable> Vm<T, VmStateFinalized> {
                             }
                         };
 
-                        // We only emulate WFI for now. Everything else gets redirected as an illegal instruction exception.
-                        match inst.instruction() {
-                            Instruction::Wfi => {
-                                // Just advance SEPC and exit. We place no constraints on when a vCPU
-                                // may be resumed from WFI since, per the privileged spec, it's only
-                                // a hint and it's perfectly valid for WFI to be a no-op.
-                                active_vcpu.inc_sepc(inst.len() as u64);
-                                break VmExitCause::Wfi;
-                            }
-                            _ => {
-                                active_vcpu.inject_exception(
-                                    Exception::IllegalInstruction,
-                                    inst.raw() as u64,
-                                );
-                                continue;
-                            }
-                        }
+                        match Self::process_decoded_instruction(&mut active_vcpu, inst) {
+                            ControlFlow::Continue(_) => continue,
+                            ControlFlow::Break(_) => break VmExitCause::Wfi,
```

Comment:
Should we be breaking with the inner cause of the `Break()`?

Reply:
Yup...had done it that way since there was only one enum, this this is future proof.